### DISCUSSION TITLE
How can I get the .coor and xsc files?

### DISCUSSION THREAD

**Discussion by xmgign** [torchmd/torchmd#21] [Q&A]
I have read the tutorial. The prmtop file comes from Amber. And how can I get the coor and xsc file?
In addition, is there any new and complicated tutorial which may contain protein-ligand complex or membrane protein system?
I noticed that torchmd has not been updated for a long time. Will it be updated in the future?

**Comment by stefdoerr** [comment]
Hi. There is no need for a `.xsc` file. The simulation runs in ~~NPT~~ NVT so the box dimensions don't change.
There is no `.coor` file but you typically write a `.npy` file. If you want to extract a frame from the `.npy` file you can do it with `moleculekit`

```python
from moleculekit.molecule import Molecule
import numpy as np

traj = np.load("./output.npy")
mol = Molecule("./structure.prmtop")
mol.coords = traj.astype(np.float32).copy()
# Write the whole traj as xtc
mol.write("./output.xtc")
# Or write a single frame as coor
mol.frame = 10 # Write the 10th frame
mol.write("./output.coor")
```

As for updates, we mostly perform bug fixes. There is not much further development planned of the code itself. We use it for applications though so it's not abandoned.

**Comment by giadefa** [comment]
Hi,
I think that Stefan meant it runs NVT simulations.

We are using torchmd internally for other projects so it is continuously
bug fixed and there will be  parts from these projects that will be
integrated into torchmd itself when ready.

**Comment by Parul32-pixe** [comment]
hi stefan,I followed this tutorial ,not able to get how to generate input.coor as it needed as input file in beginning for reading the initial simulation coordinates and same input.xsc for reading the box dimensions

**Comment by Parul32-pixe** [comment]
hi stefan,I followed this tutorial ,not able to get how to generate input.coor as it needed as input file in beginning for reading the initial simulation coordinates and same input.xsc for reading the box dimensions.

**Reply by stefdoerr** [reply]
Sorry I don't understand what the question is

**Reply by Parul32-pixe** [reply]
in the system setup we need these 2 files ,so I want to know like how can we generate these 2 files from .npy

**Reply by stefdoerr** [reply]
Can you point me to the tutorial you are referencing?

**Reply by Parul32-pixe** [reply]
![torch md](https://user-images.githubusercontent.com/84332871/199473885-c0418069-3a69-4b2e-bedf-05844621abf9.png)

**Reply by Parul32-pixe** [reply]
here these 2 files input.coor and input.xsc,this xsc file is generated by acemd3 but how can i generate the same any syntax and input.coor ,i have a doubt like is it necessary to convert coordinate file into binary file ,i extracted the coordinates from pdb file using biopython package so i can use the same here

**Reply by stefdoerr** [reply]
what files do you currently have for your system? If you have just a PDB then you should start by building the system with something like the HTMD AMBER builder https://software.acellera.com//htmd/tutorials/system-building-protein-ligand.html to generate the prmtop file

**Reply by Parul32-pixe** [reply]
thank you so much for your help and yes I do have a pdb file only in my system
